cdap-docs/developers-manual/source/building-blocks/services.rst

```diff
@@ -212,6 +212,40 @@ If both the memory and the number of cores needs to be set, this can be done usi
 An example of setting ``Resources`` using runtime arguments is shown in :ref:`Purchase
 <examples-purchase>` example's ``PurchaseHistoryBuilder.java``.
 
+.. _services-routing:
+
+Service Routing
+===============
+When multiple versions of the same service are running, you can control where service
+requests are routed.
+
+For example, if version ``v1`` and version ``v2`` of the same application are running, you
+can choose to direct 50% of the requests to version ``v1`` of the service and 50% to version
+``v2``. This can be achieved by uploading a *route configuration* (also known as a *route
+config*): a map of version names to the percentage of requests to be routed to that
+version.
+
+For a specific service, if a route config is not present or if it cannot be retrieved, a
+fallback routing strategy is used. The strategy used can be configured in the
+``cdap-site.xml`` file.
+
+These fallback strategies are available: *random*, *smallest*, *largest*, and *drop*.
+
+**Random** is the default fallback strategy. If the random fallback strategy is chosen,
+the request is routed to any version of the service.
+
+If **smallest** is chosen, the request is routed to the smallest version (based on a
+string comparison of the available versions). Similarly for **largest**: the request
+is routed to the largest version. If **drop** is chosen as the fallback strategy, the
```

Comment:
This for fall back strategy. It is expected that the route config will be available if the user is trying to route to multiple versions at the same time. So it is only in the case where we can't find the routeConfig, which is quite rare. Even in that case, "-SNAPSHOT" in version string is not recommended for production since it is only for backward compatibility.